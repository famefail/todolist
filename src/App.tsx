import Todolist from './component/todolist/Container'

function App() {
  return (
    <Todolist/>
  )
}

export default App
