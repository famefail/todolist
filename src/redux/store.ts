import { combineReducers, configureStore } from '@reduxjs/toolkit'
import todoReducer from './todo/reduceTodo'
export const rootReducer = combineReducers({
    todos: todoReducer,
  })
  
  const getTodolist = localStorage.getItem('todolist')
  let todolist = {}
  if (getTodolist) {
    todolist = JSON.parse(getTodolist)
  }

  const store = configureStore({
    reducer: rootReducer,
    preloadedState: todolist,
  })
  
  store.subscribe(() => {
    localStorage.setItem('todolist', JSON.stringify(store.getState()))
  })

  export default store
  
  export type RootState = ReturnType<typeof rootReducer>