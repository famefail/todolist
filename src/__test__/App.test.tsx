import { render } from "@testing-library/react"
import { Provider } from "react-redux"
import store from "../redux/store"
import App from "../App"
import "@testing-library/jest-dom"

describe('app', () => {
  test('container', () => {
    const { container, getByText, getByRole } = render(<Provider store={store}> <App /></Provider >)
    expect(getByText('Todolist')).toBeInTheDocument()
    expect(getByRole('button', { name: 'add' })).toBeInTheDocument()
    expect(getByRole('combobox')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})