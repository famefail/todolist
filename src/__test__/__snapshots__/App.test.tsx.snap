// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app demo 1`] = `
<div>
   
  <div
    class="bg-black"
    data-testid="todolist-container"
  >
    <header
      class="text-red-400"
    >
      Todolist
    </header>
    <input
      value=""
    />
     
    <button>
      add
    </button>
    <select
      data-testid="dropdown-container"
      id="dropdown-container"
      name="filterTodo"
    >
      <option
        value="showAll"
      >
        Show all
      </option>
      <option
        value="completed"
      >
        Completed
      </option>
      <option
        value="incomplete"
      >
        Incomplete
      </option>
    </select>
    <ul
      class=""
    />
  </div>
</div>
`;
